api/src/main/java/org/openmrs/module/attachments/AttachmentsServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package org.openmrs.module.attachments;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.Visit;
+
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.context.Context;
+
+import org.openmrs.module.attachments.obs.Attachment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AttachmentsServiceImpl implements AttachmentsService {
+	
+	private final Log log = LogFactory.getLog(getClass());
+	
+	@Autowired
+	@Qualifier(AttachmentsConstants.COMPONENT_ATT_CONTEXT)
+	private AttachmentsContext context;
+	
+	@Autowired
+	@Qualifier("conceptService")
+	private ConceptService cs;
+	
+	@Override
+	public List<Attachment> getAttachments(Patient patient, Encounter encounter, boolean includeRetired) {
+		
+		List<Person> personList = new ArrayList<>();
+		List<Attachment> attachmentsList = new ArrayList<>();
+		personList.add(patient);
+		List<String> conceptComplexList = context.getConceptComplexList();
+		List<Concept> questionConcepts = new ArrayList<>();
+		for (String uuid : conceptComplexList) {
+			Concept concept = cs.getConceptByUuid(uuid);
+			if (concept == null) {
+				log.error("The Concept with UUID " + uuid + " was not found");
+			} else {
+				questionConcepts.add(concept);
+			}
+		}
```

Comment:
At this point if the list of question concepts is empty what happens? I would assume that an `APIException` should be thrown. You could set a message like:
>No concepts complex were configured or found to query attachments.